examples/react-full-v1/pages/star-wars/index/+onBeforeRender.ts

```diff
@@ -1,11 +1,12 @@
 // https://vike.dev/onBeforeRender
 
-import type { Hook, HookParam, HookReturn } from 'vike/types'
+import type { OnBeforeRender, OnBeforeRenderParam, OnBeforeRenderReturn } from 'vike/types'
 import { filterMoviesData, getStarWarsMovies, getTitle } from './getStarWarsMovies'
 
-const onBeforeRender: Hook<'onBeforeRender'> = async (
-  pageContext: HookParam<'onBeforeRender'>
-): HookReturn<'onBeforeRender'> => {
+// NOTE(aurelien): that's the alternative style, still with full safety. I'm wondering if we should
+// keep `Config['onBeforeRender']` instead of `OnBeforeRender` in order for the user to easily modify
+// `Config` through `Vike.Config` without unexpected results?
+const onBeforeRender: OnBeforeRender = async (pageContext: OnBeforeRenderParam): OnBeforeRenderReturn => {
```

Comment:
The other reasons why this style isn't perfect are:
1. If that hook has to run on client too, you probably don't want to use `OnBeforeRenderParam` but `PageContextServer | PageContextClient` instead.
2. The direct definition of `OnBeforeRenderReturn` is still cryptic as it is `Extract<OnBeforeRenderReturnType, Promise<any>>`.

Reply:
My idea was this:

> I think a semi-satisfactory solution would be to tell users to not use `OnBeforeRender{Param,Return}` if they change the `meta` of `onBeforeRender`.

The `meta` feature is mostly meant for `vike-*` packages and, so far, I expect only few users to directly use the `meta` feature. That's why I'm thinking my semi-satisfactory proposal to be viable.